### PR TITLE
feat: add tekton v1beta1 support

### PR DESCRIPTION
**Pull request by eddycharly**
This PR adds tekton `v1beta1` support (side by side with `v1alpha1`).

### CODE REVIEW

**Comment by eddycharly** [comment]
/hold
cc @alvaroaleman

**Comment by eddycharly** [comment]
On the other hand, the new field may be better used for `v1beta1` :thinking:

**Comment by eddycharly** [comment]
cc @qaifshaikh

File: prow/apis/prowjobs/v1/types.go
Comment (qaifshaikh) [inline prow/apis/prowjobs/v1/types.go:197]: Is `PipelineRunSpec` different from `TektonPipelineRunSpec`?

Comment (qaifshaikh) [reply prow/apis/prowjobs/v1/types.go:197]: If its only for `v1beta1` then maybe we should add that in the comment

**Review by qaifshaikh — COMMENTED** [review]
I think we still need an equivalent for [run](https://github.com/kubernetes/test-infra/blob/master/prow/pipeline/informers/externalversions/pipeline/v1alpha1/run.go), right?
It was moved to a separate [api](https://github.com/tektoncd/pipeline/tree/main/pkg/apis/run/v1alpha1)

Otherwise LGTM

File: prow/apis/prowjobs/v1/types.go
Comment (eddycharly) [reply prow/apis/prowjobs/v1/types.go:197]: `TektonPipelineRunSpec` is a struct that can contain different versions (currently beta1 but we could add v1 there)

**Comment by eddycharly** [comment]
> I think we still need an equivalent for [run](https://github.com/kubernetes/test-infra/blob/master/prow/pipeline/informers/externalversions/pipeline/v1alpha1/run.go), right?

Not sure what you mean here. Isn’t Run used for custom tasks ?

**Comment by qaifshaikh** [comment]
Oh I see. I thought it was the run status API for all pipelines.

**Review by qaifshaikh — APPROVED** [review]

**Comment by qaifshaikh** [comment]
Can we move forward with this @alvaroaleman ? Its not a breaking change anymore and makes it easy to upgrade to `v1` when it becomes a standard!

**Comment by eddycharly** [comment]
We need #27032 first.
@alvaroaleman WDYT ?

**Comment by eddycharly** [comment]
Please note this PR needs a bit more work (docs, unit tests and better error handling). I will address missing bits soon, I’m currently on vacations.

**Comment by eddycharly** [comment]
Now https://github.com/kubernetes/test-infra/pull/27032 is in, I will try to terminate this one next week :crossed_fingers:

**Comment by qaifshaikh** [comment]
Hey @eddycharly any idea how long until we can get this through?

**Comment by qaifshaikh** [comment]
Any chance we can get this in before October end? 🙏🏻  @eddycharly

**Comment by Halamix2** [comment]
Any updates?

**Comment by qaifshaikh** [comment]
We were waiting on @eddycharly for this one 🤔

**Comment by eddycharly** [comment]
I was quite busy, sorry for that. I will terminate this next week !

**Comment by eddycharly** [comment]
Needs a couple more unit tests and we should be good.

**Comment by danilo-gemoli** [comment]
I'm interested in merging this one, @eddycharly how can I help?

**Comment by eddycharly** [comment]
@danilo-gemoli a couple unit tests are missing, I will add them by the end of the week (probably during the week-end).

**Comment by qaifshaikh** [comment]
would be nice to close this off before the year ends 😄

**Comment by danilo-gemoli** [comment]
I'll join @qaifshaikh eventually!

**Comment by Halamix2** [comment]
Can I help somehow?

**Comment by zhouhaibing089** [comment]
Since a few releases back in Tekton, the v1alpha1 support has been dropped totally. It would be great to see v1beta1 support (or even v1 support) in Prow.

**Comment by eddycharly** [comment]
Tomorrow I promise !!

**Comment by eddycharly** [comment]
Folks, I added the missing unit tests, please let me know how this looks.
Sorry for the long delay.

cc @stevekuznetsov @alvaroaleman

**Comment by eddycharly** [comment]
> @eddycharly: GitHub didn't allow me to request PR reviews from the following users: stevekuznetsov, alvaroaleman, chaodaiG.
> 
> Note that only [kubernetes members](https://github.com/orgs/kubernetes/people) and repo collaborators can review this PR, and authors cannot review their own PRs.

Not sure why this message 🤔

**Comment by BenTheElder** [comment]
huh, they're definitely Kubernetes members.
/cc chaodaiG

**Comment by eddycharly** [comment]
> huh, they're definitely Kubernetes members.

yeah, there's something strange here.

**Comment by BenTheElder** [comment]
This message is automatically posted when github throws a certain error making the API call to assign people. Maybe something changed in the API.

**Comment by qaifshaikh** [comment]
/lgtm 
this should simplify the `v1` upgrades as well. Really appreciate this!

**Review by stevekuznetsov — APPROVED** [review]
/hold 
for any other :eyes:

**Comment by qaifshaikh** [comment]
Are we waiting for reviews from @michelle192837 and @cjwagner ?
Would be nice to merge this soon since we have to do more work on the newer tekton pipeline version upgrades!

**Comment by michelle192837** [comment]
Don't wait up on me for review, I'm just investigating as oncall  😅 

speaking of: 

/cc michelle192837

**Comment by eddycharly** [comment]
Ok, I will remove the hold tomorrow if no one steps in until then 🤞

**Comment by michelle192837** [comment]
Sorry, last bit of spam.

/cc @michelle192837

**Pull request merged**
